langchain4j-core/src/main/java/dev/langchain4j/model/chat/response/StreamingChatResponseHandler.java

```diff
@@ -0,0 +1,47 @@
+package dev.langchain4j.model.chat.response;
+
+import dev.langchain4j.data.message.AiMessage;
+import dev.langchain4j.model.chat.StreamingChatLanguageModel;
+
+/**
+ * TODO review all javadoc in this class
+ * Represents a handler for streaming responses from a {@link StreamingChatLanguageModel}.
+ * The handler is invoked each time the model generates a new token in a textual response.
+ * If the model executes a tool instead, {@link #onCompleteResponse} will be invoked instead.
+ *
+ * @see StreamingChatLanguageModel
+ */
+public interface StreamingChatResponseHandler {
```

Comment:
Adding a default empty implementation also to the `onError` method it would be possible to annotate the `StreamingChatResponseHandler` as a `@FunctionalInterface` and use it as such allowing users to implement the `onPartialResponse` with a simple lambda or method reference as for example:
`streamingChatModel.chat(chatRequest, System.out::print);`

Reply:
Hmm, I was actually considering moving in the opposite direction and removing the default empty implementation for `onCompleteResponse`. I can't remember who wrote this, but someone was blogging about the LC4j API, and they had trouble with this listener because the IDE did not offer to implement the `onCompleteResponse` method by default. This led to confusion about why the stream did not stop.

Having a default empty implementation for `onError` might lead to the same kind of confusion. When it fails due to some problem (e.g., an incorrect API key), the default implementation will just hide it, and if the user doesn’t override it, this will cause confusion.

While convenience is definitely important, I’m not sure it should be the priority for a low-level API. WDYT?

BTW, there is a `LambdaStreamingResponseHandler` introduced by @glaforge, which could be a good compromise.